fuzz/fuzz_targets/streams.rs

```diff
@@ -0,0 +1,22 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use arbitrary::{Arbitrary};
+
+extern crate proto;
+use proto::fuzzing::{Streams};
+
+use proto::{Side};
+
+#[derive(Arbitrary, Debug)]
+struct StreamsParams {
+    side: Side,
+    max_remote_uni: u64,
+    max_remote_bi: u64,
+    send_window: u64,
+    receive_window: u64,
+    stream_receive_window: u64
+}
+
+fuzz_target!(|data: StreamsParams| {
+    Streams::new(data.side, data.max_remote_uni, data.max_remote_bi, data.send_window, data.receive_window, data.stream_receive_window);
```

Comment:
Here's just a dead simple target of `Streams`, and next week I can try to excersise some of the stream methods like `reset` etc on the control flow. 

On this target the fuzzer will discover OOM error after a few minutes; not sure how significant that is.  [I've pushed a branch on my fork](https://github.com/jafow/quinn/runs/964637707?check_suite_focus=true) that runs the fuzzer with a github action -- I expect that the CI container keeps a conservative limit on memory.

Reply:
> next week I can try to excersise some of the stream methods like reset etc on the control flow.

That would be great! I'd be particularly interested to see the various methods called in arbitrary sequence; maybe an `#[derive(Arbitrary)] enum Operation { Open, Reset(...), ... }` or similar?

> On this target the fuzzer will discover OOM error after a few minutes; not sure how significant that is

I'm guessing this happens when `max_remote_uni` or `max_remote_bi` are very large, since we preallocate state proportional to those. We should probably give those a modest maximum value since the number of streams shouldn't change behavior much at the margin. A quick hack to do that would be using `u8` instead of `u64` for the relevant fields in `StreamsParams`.